.github/workflows/integrationTest.yml

```diff
@@ -14,18 +14,73 @@ env:
   PASSPHRASE: ${{ secrets.PASSPHRASE }}
   GPG_KEY_NAME: ${{ secrets.GPG_KEY_NAME }}
   GPG_TTY: $(tty)
+  ECR_INTEGRATION_TEST_REPO: "cwagent-opensource-integration-test"
 
 on:
   push:
     branches:
       - master
+    
   workflow_dispatch:
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref_name }}
   cancel-in-progress: true
 
 jobs:
+  MakeDockerImage:
+    name: 'MakeDockerImage'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Cache if success
+        id: build-docker-image
+        uses: actions/cache@v2
+        with:
+          path: |
+            RELEASE_NOTES
+          key: build-docker-image-${{ github.run_id }}
+
+      - name: Configure AWS Credentials
+        if: steps.build-docker-image.outputs.cache-hit != 'true'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.TERRAFORM_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.TERRAFORM_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+
+      - name: Login ECR
+        if: steps.build-docker-image.outputs.cache-hit != 'true'
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Set up Docker Buildx
+        if: steps.build-docker-image.outputs.cache-hit != 'true'
+        uses: docker/setup-buildx-action@v1
+
+      - name: Set up QEMU
```

Comment:
Can you explain what this is for?

Reply:
Yes, based on the [documents](https://martin-grigorov.medium.com/building-linux-packages-for-different-cpu-architectures-with-docker-and-qemu-d29e4ebc9fa5), 
```
QEMU emulates the machine’s processor through dynamic binary translation and provides a set of different hardware and device models for the machine, enabling it to run a variety of guest operating systems.
```
In another way of speaking, QEMU helps us in building multi platforms such as [arm64,amd64](https://github.com/aws/amazon-cloudwatch-agent/pull/458/files#diff-9b9f9a38b889c9dcb55ba0330baa69c8fdd62ac1100b6eb59b6fc44008f5d320R83) for our docker image.

Reply:
This is required for buildx. Why not sure. We learned this was trail and error in adot.

Reply:
I remember reading a medium doc about if `docker buildx` does not have QEMU or binfmt, it would only support `linux/amd64`; let's me try to find those documents since I have read those months ago.

Reply:
Fair enough